modules/fpga.c

```diff
@@ -66,14 +66,14 @@ STATIC mp_obj_t fpga_write(mp_obj_t addr_16bit, mp_obj_t bytes)
     uint16_t addr = mp_obj_get_int(addr_16bit);
     uint8_t addr_bytes[2] = {(uint8_t)(addr >> 8), (uint8_t)addr};
 
-    if (n == 0)
+    if (buffer.len == 0)
     {
         monocle_spi_write(FPGA, addr_bytes, 2, false);
     }
     else
     {
         monocle_spi_write(FPGA, addr_bytes, 2, true);
-        monocle_spi_write(FPGA, (uint8_t *)buffer, n, false);
+        monocle_spi_write(FPGA, buffer.buf, buffer.len, false);
```

Comment:
Not sure you added this for clarity or if because buf was `(char *)` in the past.

If you meant to keep it like that, let me know and I would adjust it back.

Reply:
This change looks good :)